docs/src/references/mappers/c8y-mapper.md

```diff
@@ -568,6 +568,305 @@ c8y/alarm/alarms/create
 
 </div>
 
+### Data
+
+The data telemetry type can be used to store additional information about the device/child/service. Such information could included; operation system name/version, communication statistics, device status, or any other information that is not suited to measurements, events or alarms.
+
+#### Data - Main device
+
+A device's digital twin model can be updated by publishing to a specific topic.
+
+The type part of the topic is used to group the data so it is easier for components to subscribe to relevant parts.
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main///data/device_OS
+```
+
+```json5 title="Payload"
+{
+  "family": "Debian",
+  "version": "11"
+}
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>
+```
+
+```json5 title="Payload"
+{
+  "device_OS": {
```

Comment:
Shouldn't we do some mapping of "special keys" in the payload as well? For example, for something like hardware info: a clous-agnostic user might use something like:

```json
"hardware": {
  "model": "BCM2708",
  "revision": "000e",
  "serialNumber": "00000000e2f5ad4d"
}
```

which needs to be mapped the following `c8y_Hardware` fragment, which Cumulocity understands:

```json
"c8y_Hardware": {
  "model": "BCM2708",
  "revision": "000e",
  "serialNumber": "00000000e2f5ad4d"
}
```

We can definitely keep "special payload key mapping" out of scope initially and then add it later if there is interest. In the mean time, users of C8y are free to send the `c8y_Hardware` fragment directly.